src/api/IInput.js

```diff
@@ -56,5 +56,29 @@
     IInput.prototype.change = function (w) {
     };
 
+    IInput.prototype.checkNodeName = function (expected, element) {
+        var remove = this._inputElement.some(function(e) {
+            if (e.node().nodeName !== expected) {
+                return true;
+            }
+        });
+        if (remove) {
+            this._inputElement.forEach(function(e) {
+               e.node().remove();
+            });
+            this.createInput(element);
+        }
```

Comment:
@GordonSmith Originally, yes, the Input widget was built with the intention of an "input" being able to change "type" (ex: from a "text" input to a textarea).

But I don't think this concept works anymore with the latest implementation of form inputs (I don't think its necessary).